plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ScanQueryPageSource.java

```diff
@@ -272,7 +273,7 @@ private void flattenFields(Map<String, Type> result, String fieldName, Type type
                 .collect(toImmutableList());
     }
 
-    private Decoder createDecoder(String path, Type type)
+    public static Decoder createDecoder(String path, Type type)
```

Comment:
Instead of making this method public and introducing a spurious depedency between AggregationQueryPageSource and this class, this method should move somewhere else. E.g., to a new class named `Decoders`